pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java

```diff
@@ -0,0 +1,210 @@
+
+public class PravegaSensorCollectorIntegrationTests {
+    private static final Logger log = LoggerFactory.getLogger(PravegaSensorCollectorIntegrationTests.class);
+    private final SetupUtils setupUtils = new SetupUtils();
+    static String fileName = "./src/test/resources/RawFileIngest-integration-test.properties";
+    Map<String, String> properties = null;
+    @BeforeEach
+    public void setup() {
+        log.info("Setup");
+        properties = Parameters.getProperties(fileName);
+        try {
+            setupUtils.startAllServices();
+
+            Files.deleteIfExists(Paths.get(properties.get("PRAVEGA_SENSOR_COLLECTOR_RAW1_DATABASE_FILE")));
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        properties.put("PRAVEGA_SENSOR_COLLECTOR_RAW1_PRAVEGA_CONTROLLER_URI", setupUtils.getControllerUri().toString());
+        log.debug("Properties: {}", properties);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        log.info("TearDown");
+        try {
+            setupUtils.stopAllServices();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+
+        try {
+            Files.deleteIfExists(Paths.get(properties.get("PRAVEGA_SENSOR_COLLECTOR_RAW1_DATABASE_FILE")));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        properties = null;
+    }
+
+    @Test
+    @Timeout(value = 3, unit = TimeUnit.MINUTES)
+    public void testPSCDataIntegration() {
+        try {
+            copyHelloWorldFile();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        URI controllerURI = setupUtils.getControllerUri();
+        String scope = "test-psc-data-integration";
+        String streamName ="test-psc-data-integration-stream";
+
+        properties.put("PRAVEGA_SENSOR_COLLECTOR_RAW1_SCOPE",scope);
+        properties.put("PRAVEGA_SENSOR_COLLECTOR_RAW1_STREAM",streamName);
+
+        final DeviceDriverManager deviceDriverManager = new DeviceDriverManager(properties);
+        Service startService = deviceDriverManager.startAsync();
+        try {
+            startService.awaitRunning(Duration.ofSeconds(30));
+            Thread.sleep(25000);
+        } catch (InterruptedException | TimeoutException e) {
+            throw new RuntimeException(e);
+        }
+        final Connection connection = SQliteDBUtility.createDatabase(properties.get("PRAVEGA_SENSOR_COLLECTOR_RAW1_DATABASE_FILE"));
+        final TransactionStateDB state = new TransactionStateSQLiteImpl(connection, null);
+
+        try {
+            List<FileNameWithOffset> completedFiles = state.getCompletedFileRecords();
+            Assertions.assertEquals(1, completedFiles.size());
+
+            validateStreamData(controllerURI, scope, streamName, new String(Files.readAllBytes(Paths.get("../parquet-file-sample-data/test_file/hello-world.parquet"))));
+
+            Thread.sleep(50000);
```

Comment:
Question: can we remove this Thread.sleep by reducing the delete interval for this test alone? is that configurable? And then by the time line 108 finishes, we can either assert or assertEventually on completedFiles size being 0

Reply:
The min value is 1 minute, which I seted up for these tests.

Reply:
reduced sleep interval to 5 seconds from 50 seconds.